src/rapids_pre_commit_hooks/copyright.py

```diff
@@ -0,0 +1,230 @@
+
+
+def strip_copyright(content, copyright_matches):
+    lines = []
+
+    def append_stripped(start, item):
+        lines.append(content[start : item.start()])
+        return item.end()
+
+    start = functools.reduce(append_stripped, copyright_matches, 0)
+    lines.append(content[start:])
+    return lines
+
+
+def apply_copyright_check(linter, old_content):
+    if linter.content != old_content:
+        current_year = datetime.datetime.now().year
+        new_copyright_matches = match_copyright(linter.content)
+
+        if old_content is not None:
+            old_copyright_matches = match_copyright(old_content)
+
+        if old_content is not None and strip_copyright(
+            old_content, old_copyright_matches
+        ) == strip_copyright(linter.content, new_copyright_matches):
+            for old_match, new_match in zip(
+                old_copyright_matches, new_copyright_matches
+            ):
+                if old_match.group() != new_match.group():
+                    if old_match.group("years") == new_match.group("years"):
+                        warning_pos = new_match.span()
+                    else:
+                        warning_pos = new_match.span("years")
+                    linter.add_warning(
+                        warning_pos,
+                        "copyright is not out of date and should not be updated",
+                    ).add_replacement(new_match.span(), old_match.group())
+        else:
+            if new_copyright_matches:
+                for match in new_copyright_matches:
+                    if (
+                        int(match.group("last_year") or match.group("first_year"))
+                        < current_year
+                    ):
+                        linter.add_warning(
+                            match.span("years"), "copyright is out of date"
+                        ).add_replacement(
+                            match.span(),
+                            f"Copyright (c) {match.group('first_year')}-{current_year}"
+                            ", NVIDIA CORPORATION",
+                        )
+            else:
+                linter.add_warning((0, 0), "no copyright notice found")
+
+
+def get_target_branch(repo):
+    # Try environment
+    target_branch_name = os.getenv("GITHUB_BASE_REF")
+    if target_branch_name:
+        try:
+            return repo.heads[target_branch_name]
+        except IndexError:
+            pass
+    target_branch_name = os.getenv("TARGET_BRANCH")
+    if target_branch_name:
+        try:
+            return repo.heads[target_branch_name]
+        except IndexError:
+            pass
+    target_branch_name = os.getenv("RAPIDS_BASE_BRANCH")
+    if target_branch_name:
+        try:
+            return repo.heads[target_branch_name]
+        except IndexError:
+            pass
+
+    # Try config
+    with repo.config_reader() as r:
+        target_branch_name = r.get("rapidsai", "baseBranch", fallback=None)
```

Comment:
Having some configuration for the behavior of this function would be nice. I know https://github.com/NVIDIA/cuCollections/ would like to use this hook, too. The base branch detection would be much simpler (it's always called `dev`, I think).